resources/sts/4.14/hypershift/sts_extended_hcp_support_permission_policy.json

```diff
@@ -2,58 +2,133 @@
+            "Action": [
                 "ec2:DescribeClientVpnAuthorizationRules",
                 "ec2:DescribeClientVpnConnections",
-                "ec2:DescribeClientVpnEndpoints",
                 "ec2:DescribeClientVpnRoutes",
-                "ec2:DescribeClientVpnTargetNetworks",
-                "ec2:DescribeCoipPools",
-                "ec2:DescribeCustomerGateways",
-                "ec2:DescribeEgressOnlyInternetGateways",
-                "ec2:DescribeIdentityIdFormat",
-                "ec2:DescribeIdFormat",
-                "ec2:DescribeImageAttribute",
-                "ec2:DescribeImages",
+                "ec2:DescribeClientVpnTargetNetworks"
+            ],
+            "Resource": "arn:aws:ec2:*:*:client-vpn-endpoint/*"
+        },
+        {
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeClientVpnEndpoints"
+            ],
+            "Resource": "*"
+        },
+        {
+
+            "Sid": "Vpn",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeVpnConnections",
+                "ec2:DescribeVpnGateways"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "InstanceAttribute",
+            "Effect": "Allow",
+            "Action": [
                 "ec2:DescribeInstanceAttribute",
-                "ec2:DescribeInstanceTypeOfferings",
-                "ec2:DescribeInternetGateways",
-                "ec2:DescribeIpv6Pools",
-                "ec2:DescribeKeyPairs",
-                "ec2:DescribeLaunchTemplates",
+                "ec2:ModifyInstanceAttribute"
+            ],
+            "Resource": "arn:aws:ec2:*:*:instance/*",
+            "Condition": {
+                "StringEquals": {
+                    "aws:ResourceTag/red-hat-managed": "true"
+                }
+            }
+        },
+        {
+            "Sid": "LocalGateway",
+            "Effect": "Allow",
+            "Action": [
                 "ec2:DescribeLocalGatewayRouteTables",
                 "ec2:DescribeLocalGatewayRouteTableVirtualInterfaceGroupAssociations",
                 "ec2:DescribeLocalGatewayRouteTableVpcAssociations",
                 "ec2:DescribeLocalGateways",
                 "ec2:DescribeLocalGatewayVirtualInterfaceGroups",
-                "ec2:DescribeLocalGatewayVirtualInterfaces",
-                "ec2:DescribeManagedPrefixLists",
-                "ec2:DescribeNetworkInsightsAnalyses",
-                "ec2:DescribeNetworkInsightsPaths",
-                "ec2:DescribePlacementGroups",
-                "ec2:DescribePrefixLists",
-                "ec2:DescribePrincipalIdFormat",
-                "ec2:DescribePublicIpv4Pools",
-                "ec2:DescribeSnapshotAttribute",
-                "ec2:DescribeSnapshots",
-                "ec2:DescribeSpotFleetInstances",
-                "ec2:DescribeTags",
+                "ec2:DescribeLocalGatewayVirtualInterfaces"
+            ],
+            "Resource": "*"
```

Comment:
Can these be restricted at all?

Reply:
since localGateway is customer owned, doesn't seem like we can restrict it based on tag